src/sentry/models/groupassignee.py

```diff
@@ -18,7 +21,92 @@
 from sentry.signals import issue_assigned
 
 
-def sync_group_assignee(group, user_id, assign=True):
+def get_user_project_ids(users):
+    """
+    Given a list of users, return a dict where keys are user_ids
+    and values are a set of the project_ids the user is a member of
+    """
+    from sentry.models import OrganizationMemberTeam, ProjectTeam
+    user_teams = list(OrganizationMemberTeam.objects.filter(
+        organizationmember__user__in=users,
+        is_active=True,
+    ).values('organizationmember__user', 'team'))
+
+    # team_id to list of projects
+    projects_by_team = defaultdict(set)
+    for tp in ProjectTeam.objects.filter(team__in=[ut['team'] for ut in user_teams]):
+        projects_by_team[tp.team_id].add(tp.project_id)
+
+    # user_id to projects
+    projects_by_user = defaultdict(set)
+    for ut in user_teams:
+        projects_by_user[ut['organizationmember__user']].update(projects_by_team[ut['team']])
+
+    return projects_by_user
+
+
+def sync_group_assignee_inbound(integration, email, external_issue_key, assign=True):
+    """
+    Given an integration, user email address and an external issue key,
+    assign linked groups to matching users. Checks project membership.
+    Returns a list of groups that were successfully assigned.
+    """
+    from sentry.models import Group, GroupLink, ExternalIssue, UserEmail, User
+
+    logger = logging.getLogger('sentry.integrations.%s' % integration.provider)
+
+    affected_groups = Group.objects.filter(
```

Comment:
Maybe being a little paranoid, but can we also filter `Group` down on `organization_id` as well? Just to make sure we don't have wires crossed anywhere. I'd prefer to be overly conservative.